src/cartesian_communicator.cpp

```diff
@@ -10,12 +10,13 @@
 #include <cassert>
 
 #include <boost/mpi/cartesian_communicator.hpp>
+#include <boost/mpi/detail/antiques.hpp>
 
 namespace boost { namespace mpi {
 
 namespace {
   template <typename T, typename A>
-  T* c_data(std::vector<T,A>& v) { return &(v[0]); }
+  T* c_data(std::vector<T,A>& v) { return c_data(v); }
```

Comment:
Isn't this going to call itself recursively forever?
```suggestion
  T* c_data(std::vector<T,A>& v) { return detail::c_data(v); }
```

Reply:
It might be simpler just to replace this function with `using detail::c_data;`

Reply:
> Isn't this going to call itself recursively forever?

A legitimate concern.

Reply:
This not a recursive call, it's `boost::mpi::<anonymous namespace>::c_data` calling `boost::mpi::detail::c_data`. But I'll remove the former.

Reply:
How is it going to call `detail::c_data`? Name lookup inside the function finds that function itself, not one that isn't even visible.

Reply:
Ah sorry, I confused your suggested change with the diff. You are right of course.